gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/db/FabricDatabase.java

```diff
@@ -1453,17 +1447,15 @@ private void clearAllIndexes(ArrayList<GemFireContainer> uninitializedContainers
   private void recreateAllLocalIndexes(final LogWriter logger) {
     Collection<DiskStoreImpl> diskStores = Misc.getGemFireCache().listDiskStores();
     for (DiskStoreImpl ds : diskStores) {
-      if (!ds.getName().equals(GfxdConstants.GFXD_DD_DISKSTORE_NAME)) {
-        PersistentOplogSet oplogSet = ds.getPersistentOplogSet(null);
-        ds.resetIndexRecoveryState();
-        // delete all idx file of all oplogs, so second arg as true below
-        ds.scheduleIndexRecovery(oplogSet.getSortedOplogs(), true);
-        logger.info("FabricDatabase: recreateAllLocalIndexes " +
-            "waiting for index re-creation for disk store: " + ds.getName());
-        ds.waitForIndexRecoveryEnd(-1);
-        logger.info("FabricDatabase: recreateAllLocalIndexes " +
-            "index re-creation for disk store: " + ds.getName() + " ended");
-      }
+      PersistentOplogSet oplogSet = ds.getPersistentOplogSet(null);
```

Comment:
Yes hive metatables create indexes. These are still excluded from "persist-indexes" since the datadictionary diskstore is marked "isForInternalUse" but index recreation including consistency check should be done for that to be on the safe side (and recreate index if mismatch).